content/blog/tidymodels-2023-q4/index.Rmd

```diff
@@ -0,0 +1,179 @@
+    thanks = map_chr(repo, return_tidy_thanks, from = since),
+    thanks = glue("- {package}: {thanks}"),
+    news = glue("- {package} [({version})](https://{package}.tidymodels.org/news/index.html)")
+  )
+
+txt_pkg_list <- knitr::combine_words(tm_data$package)
+```
+
+The [tidymodels](https://www.tidymodels.org/) framework is a collection of R packages for modeling and machine learning using tidyverse principles.
+
+Since the beginning of 2021, we have been publishing [quarterly updates](https://www.tidyverse.org/categories/roundup/) here on the tidyverse blog summarizing what's new in the tidymodels ecosystem. The purpose of these regular posts is to share useful new features and any updates you may have missed. You can check out the [`tidymodels` tag](https://www.tidyverse.org/tags/tidymodels/) to find all tidymodels blog posts here, including our roundup posts as well as those that are more focused, like this post from the past couple of months:
+
+-   [Three ways errors are about to get better in tidymodels](https://www.tidyverse.org/blog/2023/11/tidymodels-errors-q4/)
+
+Since [our last roundup post](https://www.tidyverse.org/blog/2022/12/tidymodels-2022-q4/), there have been CRAN releases of `r nrow(tm_data)` tidymodels packages. Here are links to their NEWS files:
+
+```{r}
+#| echo: FALSE
+#| results: asis
+
+cat(tm_data$news, sep = "\n")
+```
+
+We'll highlight a few especially notable changes below: Updated warnings when normalizing, and better error messages in recipes.
+
+```{r}
+#| message: FALSE
+library(tidymodels)
+
+data("ames", package = "modeldata")
+```
+
+## Updated warnings when normalizing
+
+The latest release of recipes features an overhaul of the warnings and error messages to use the [cli](https://cli.r-lib.org/) package. With this, we are starting the project of providing more information signaling when things don't go well.
+
+The first type of issue we now signal for is when you try to normalize data that contains non-numeric numbers such as `NA` or `Inf`. These can sneak in for several reasons, and before this release, it happened silently. Below we are creating a recipe using the `ames` data set, and before we normalize we are taking the logarithms root of all variables that pertain to square footage. 
+
+```{r}
+rec <- recipe(Sale_Price ~ ., data = ames) |>
+  step_log(contains("SF")) |>
+  step_normalize(all_numeric_predictors()) |>
+  prep()
+```
+
+We now get a warning that something happened. Telling us that it encountered `Inf` or `-Inf`. Knowing that we can go back and investigate what went wrong. If we exclude `step_normalize()` and `bake()` the recipe, we see that a number of `-Inf` appears
+
+```{r}
+recipe(Sale_Price ~ ., data = ames) |>
+  step_log(contains("SF")) |>
+  prep() |>
+  bake(new_data = NULL, contains("SF")) |>
+  glimpse()
+```
+
+Looking at the bare data set, we notice that the `-Inf` all appear where there are `0`, which makes sense since `log(0)`.
```

Comment:
```suggestion
Looking at the bare data set, we notice that the `-Inf` all appear where there are `0`, which makes sense since `log(0)` is undefined.
```